openslides/users/static/js/users/site.js

```diff
@@ -1397,17 +1397,21 @@ angular.module('OpenSlidesApp.users.site', [
 .controller('userMenu', [
     '$scope',
     '$http',
+    '$q',
     'DS',
-    'User',
-    'operator',
+    'autoupdate',
+    'OSBootup',
     'ngDialog',
     'UserProfileForm',
     'UserPasswordForm',
-    function($scope, $http, DS, User, operator, ngDialog, UserProfileForm, UserPasswordForm) {
+    function($scope, $http, $q, DS, autoupdate, OSBootup, ngDialog, UserProfileForm, UserPasswordForm) {
         $scope.logout = function () {
             $http.post('/users/logout/').then(function (response) {
-                operator.setUser(null);
-                window.location.reload();
+                // Success: User logged out.
+                // Clear store and reset deferred first message, then do the OS-bootup process again.
+                DS.clear();
```

Comment:
Hm. What happens to the open ws connection? What if in between any new ws data come in? I think we should implement a clear reboot in core app instead of hacking some things (like the deferred and DS.clear()) here.